### PR TITLE
Implement create borrowing endpoint

### PR DESCRIPTION
**Pull request by motivated2**

### CODE REVIEW

File: borrowings/serializers.py [hunk 1]
```diff
@@ -32,3 +32,25 @@ class Meta:
             "book",
             "user"
         )
+
+
+class BorrowingCreateSerializer(serializers.ModelSerializer):
+    class Meta:
```
Comment (aLEKS-e3) [inline borrowings/serializers.py:38]: add a blank line

File: borrowings/serializers.py [hunk 2]
```diff
@@ -32,3 +32,25 @@ class Meta:
...
+        if value.inventory <= 0:
+            raise serializers.ValidationError("Book inventory is 0. Cannot borrow.")
+        return value
+
+    def create(self, validated_data):
+        book = validated_data["book"]
+        borrowing = Borrowing.objects.create(
+            expected_return_date=validated_data["expected_return_date"],
+            book=book,
+            user=self.context["request"].user
```
Comment (aLEKS-e3) [inline borrowings/serializers.py:52]: why didn't you make `self.request.user`?

**Review by aLEKS-e3 — CHANGES_REQUESTED** [review]

File: borrowings/serializers.py [hunk 3]
```diff
@@ -32,3 +32,25 @@ class Meta:
...
+
+class BorrowingCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Borrowing
+        fields = ["expected_return_date", "book"]
+
+    def validate_book(self, value):
+        if value.inventory <= 0:
+            raise serializers.ValidationError("Book inventory is 0. Cannot borrow.")
+        return value
```
Comment (aLEKS-e3) [inline borrowings/serializers.py:45]: cover validation with a test

**Review by aLEKS-e3 — CHANGES_REQUESTED** [review]

**Review by kabust — APPROVED** [review]
GJ!

File: borrowings/tests/test_serializers.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+class BorrowingCreateSerializerTest(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.test", password="test_password"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_create_borrowing_with_invalid_data(self):
+        book = Book.objects.create(title='Test Book', author='Test Author', cover='HARD', inventory=0, daily_fee=10.0)
```
Comment (aLEKS-e3) [inline borrowings/tests/test_serializers.py:22]: split this line it's too long

File: borrowings/tests/test_serializers.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            'expected_return_date': '2024-04-05',
+            'book': book.pk
+        }
+
+        res = self.client.post(BORROWINGS_URL, payload)
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_create_borrowing_with_valid_data(self):
+        book = Book.objects.create(title='Test Book', author='Test Author', cover='HARD', inventory=10, daily_fee=10.0)
```
Comment (aLEKS-e3) [inline borrowings/tests/test_serializers.py:33]: and this one

**Review by aLEKS-e3 — CHANGES_REQUESTED** [review]

File: borrowings/views.py [hunk 6]
```diff
@@ -14,4 +14,9 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action in ("list", "retrieve"):
             return BorrowingListSerializer
+        if self.action == "create":
+            return BorrowingCreateSerializer
         return self.serializer_class
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
```
Comment (aLEKS-e3) [inline borrowings/views.py:22]: add a blank line in the end

**Review by aLEKS-e3 — CHANGES_REQUESTED** [review]

**Review by aLEKS-e3 — APPROVED** [review]

**Pull request merged**